# Movie Price Data from King County

Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


## Data Science Lifecycle
[![DataScienceLifecycle](https://lh3.googleusercontent.com/proxy/a9eWy4BY9AjowoS_OJqNr84YhRysDCPQiF1-K1SnwdoyJYqiwiw4wP7nY5yc_MXKBN2_MsY96TXSeOkHKkHr4mX8rd_lhQL2masR7CC0R0ud3A2-rCY8E56vzPRt3a1Aap5IjcREfUaNNl84WWa5yV0NcGuiq7smXA)](https://lh3.googleusercontent.com/proxy/a9eWy4BY9AjowoS_OJqNr84YhRysDCPQiF1-K1SnwdoyJYqiwiw4wP7nY5yc_MXKBN2_MsY96TXSeOkHKkHr4mX8rd_lhQL2masR7CC0R0ud3A2-rCY8E56vzPRt3a1Aap5IjcREfUaNNl84WWa5yV0NcGuiq7smXA)

We structure this project based off the above Data Science Lifecyle to structure and organize this project. In this project we will follow this structure. We will see, at times, these steps will be revisited.

## Business  Understanding
From above, we see at "01" that we must obtain ask ourselves relevant questions and define objectives:
### Questions:
* "Why did you pick those features as predictors?"
* "How confident are you in the predictive quality of the results?"
* "What are some of the things that could cause the results to be wrong?"

### Objectives:
* Clean, explore, and model this dataset with a multivariate linear regression to predict the sale price of houses as accurately as possible
* Render at least 4 meaningful data visualizations, with corresponding interpretations
* Pose at least 3 meaningful questions and answer them through EDA
* 1 - 2 paragraphs briefly explaining your approach to this project
* Model should not include any predictors with p-values greater than .05.
* An iterative approach to modeling, and details the parameters and results of the model at each iteration
* Provide at least 1 paragraph explaining your final model.
* Pick at least 3 coefficients from your final model and explain their impact on the price of a house in this dataset.
* Identify at least two concrete features that highly influence housing prices.

## Data Mining
Moving on to "02", we must gather and scrape the data necessary for this project. We will only be using data from King County dataset provided by Flatiron School. We import this data into a pandas dataframe and display the first five entries to ensure correct importation:

### Importing Data

In [2]:
import pandas as pd

df = pd.read_csv('./kc_house_data.csv')
df.head()

id        date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  10/13/2014  221900.0         3       1.00         1180   
1  6414100192   12/9/2014  538000.0         3       2.25         2570   
2  5631500400   2/25/2015  180000.0         2       1.00          770   
3  2487200875   12/9/2014  604000.0         4       3.00         1960   
4  1954400510   2/18/2015  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0         NaN   0.0  ...      7        1180            0.0   
1      7242     2.0         0.0   0.0  ...      7        2170          400.0   
2     10000     1.0         0.0   0.0  ...      6         770            0.0   
3      5000     1.0         0.0   0.0  ...      7        1050          910.0   
4      8080     1.0         0.0   0.0  ...      8        1680            0.0   

  yr_built  yr_renovated  zipcode      lat     long  sqft_living15  sqft_lot15  
0     1955           0.0    98178  47.5112 -122.257           1340        5650  
1     1951        1991.0    98125  47.7210 -122.319           1690        7639  
2     1933           NaN    98028  47.7379 -122.233           2720        8062  
3     1965           0.0    98136  47.5208 -122.393           1360        5000  
4     1987           0.0    98074  47.6168 -122.045           1800        7503  

[5 rows x 21 columns]

## Data Cleaning
Next, "03" says we fix inconsistencies within our data and handle missing values. Inconsistencies mean for example the presence of extremely high or low numbers which are code for absent data or just plain NaN values (i.e. also means absent data). Let us start looking:

In [5]:
df.info() # helps us see in which columns the NaN's reside

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
id               21597 non-null int64
date             21597 non-null object
price            21597 non-null float64
bedrooms         21597 non-null int64
bathrooms        21597 non-null float64
sqft_living      21597 non-null int64
sqft_lot         21597 non-null int64
floors           21597 non-null float64
waterfront       19221 non-null float64
view             21534 non-null float64
condition        21597 non-null int64
grade            21597 non-null int64
sqft_above       21597 non-null int64
sqft_basement    21597 non-null object
yr_built         21597 non-null int64
yr_renovated     17755 non-null float64
zipcode          21597 non-null int64
lat              21597 non-null float64
long             21597 non-null float64
sqft_living15    21597 non-null int64
sqft_lot15       21597 non-null int64
dtypes: float64(8), int64(11), object(2)
memory usage: 3.5+ MB


From above, we see that most columns have 21597 non-null entires. Let's look at the stats for each column as well:

In [6]:
df.describe() #helps us see how skewed the data is by comparing the column's mean to the column's 50% quartile.

id         price      bedrooms     bathrooms   sqft_living  \
count  2.159700e+04  2.159700e+04  21597.000000  21597.000000  21597.000000   
mean   4.580474e+09  5.402966e+05      3.373200      2.115826   2080.321850   
std    2.876736e+09  3.673681e+05      0.926299      0.768984    918.106125   
min    1.000102e+06  7.800000e+04      1.000000      0.500000    370.000000   
25%    2.123049e+09  3.220000e+05      3.000000      1.750000   1430.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.159700e+04  21597.000000  19221.000000  21534.000000  21597.000000   
mean   1.509941e+04      1.494096      0.007596      0.233863      3.409825   
std    4.141264e+04      0.539683      0.086825      0.765686      0.650546   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068500e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above      yr_built  yr_renovated       zipcode  \
count  21597.000000  21597.000000  21597.000000  17755.000000  21597.000000   
mean       7.657915   1788.596842   1970.999676     83.636778  98077.951845   
std        1.173200    827.759761     29.375234    399.946414     53.513072   
min        3.000000    370.000000   1900.000000      0.000000  98001.000000   
25%        7.000000   1190.000000   1951.000000      0.000000  98033.000000   
50%        7.000000   1560.000000   1975.000000      0.000000  98065.000000   
75%        8.000000   2210.000000   1997.000000      0.000000  98118.000000   
max       13.000000   9410.000000   2015.000000   2015.000000  98199.000000   

                lat          long  sqft_living15     sqft_lot15  
count  21597.000000  21597.000000   21597.000000   21597.000000  
mean      47.560093   -122.213982    1986.620318   12758.283512  
std        0.138552      0.140724     685.230472   27274.441950  
min       47.155900   -122.519000     399.000000     651.000000  
25%       47.471100   -122.328000    1490.000000    5100.000000  
50%       47.571800   -122.231000    1840.000000    7620.000000  
75%       47.678000   -122.125000    2360.000000   10083.000000  
max       47.777600   -121.315000    6210.000000  871200.000000

From the above, we see that 'waterfront', 'view', and 'yr_renovated' have NaN values. Let us look at 'waterfront' first, then we will look at the others. 'waterfront' has a very low mean of 0.007596. Let's take a closer look at it.

In [11]:
df.waterfront.value_counts() #this gives what values are in column

0.0    19075
1.0      146
Name: waterfront, dtype: int64

In [12]:
len(df[df.waterfront.isna()]) #this tells us how many values are NaN out of 21597

2376

In [27]:
len(df.waterfront.values) #makes sure there are 21597 entries

21597

In [21]:
#Percent that is either 0 (i.e.'No') or NaN from waterfront
print(f'Percent that is either 0 (i.e.\'No\') or NaN: {round((df.waterfront.value_counts()[0]+len(df[df.waterfront.isna()]))/len(df) * 100, 2)}%')

Percent that is either 0 (i.e.'No') or NaN: 99.32%


From the above, we determine that we can drop waterfront, because virtually all the houses do not have a 'waterfront'. Before we do this, let us check out if we can drop 'view' and 'yr_renovated' as well.

In [22]:
df.view.value_counts()

0.0    19422
2.0      957
3.0      508
1.0      330
4.0      317
Name: view, dtype: int64

In [23]:
len(df[df.view.isna()])

63

In [26]:
len(df.view.values)

21597

In [28]:
print(f'Percent that is either 0 (i.e.\'No\') or NaN: {round((df.view.value_counts()[0]+len(df[df.view.isna()]))/len(df) * 100, 2)}%')

Percent that is either 0 (i.e.'No') or NaN: 90.22%


In [62]:
len(df.id.unique())

21420

In [111]:
df[df.duplicated(subset='id')]

id        date      price  bedrooms  bathrooms  sqft_living  \
94     6021501535  12/23/2014   700000.0         3       1.50         1580   
314    4139480200   12/9/2014  1400000.0         4       3.25         4290   
325    7520000520   3/11/2015   240500.0         2       1.00         1240   
346    3969300030  12/29/2014   239900.0         4       1.00         1000   
372    2231500030   3/24/2015   530000.0         4       2.25         2180   
...           ...         ...        ...       ...        ...          ...   
20165  7853400250   2/19/2015   645000.0         4       3.50         2910   
20597  2724049222   12/1/2014   220000.0         2       2.50         1000   
20654  8564860270   3/30/2015   502000.0         4       2.50         2680   
20764  6300000226    5/4/2015   380000.0         4       1.00         1200   
21565  7853420110    5/4/2015   625000.0         3       3.00         2780   

       sqft_lot  floors  waterfront  view  ...  grade  sqft_above  \
94         5000     1.0         0.0   0.0  ...      8        1290   
314       12103     1.0         0.0   3.0  ...     11        2690   
325       12092     1.0         0.0   0.0  ...      6         960   
346        7134     1.0         0.0   0.0  ...      6        1000   
372       10754     1.0         0.0   0.0  ...      7        1100   
...         ...     ...         ...   ...  ...    ...         ...   
20165      5260     2.0         0.0   0.0  ...      9        2910   
20597      1092     2.0         0.0   0.0  ...      7         990   
20654      5539     2.0         NaN   0.0  ...      8        2680   
20764      2171     1.5         0.0   0.0  ...      7        1200   
21565      6000     2.0         0.0   0.0  ...      9        2780   

       sqft_basement yr_built  yr_renovated  zipcode      lat     long  \
94             290.0     1939           0.0    98117  47.6870 -122.386   
314           1600.0     1997           0.0    98006  47.5503 -122.102   
325            280.0     1922        1984.0    98146  47.4957 -122.352   
346              0.0     1943           NaN    98178  47.4897 -122.240   
372           1080.0     1954           0.0    98133  47.7711 -122.341   
...              ...      ...           ...      ...      ...      ...   
20165            0.0     2012           0.0    98065  47.5168 -121.883   
20597           10.0     2004           0.0    98118  47.5419 -122.271   
20654            0.0     2013           0.0    98045  47.4759 -121.734   
20764            0.0     1933           0.0    98133  47.7076 -122.342   
21565            0.0     2013           NaN    98065  47.5184 -121.886   

       sqft_living15  sqft_lot15  
94              1570        4500  
314             3860       11244  
325             1820        7460  
346             1020        7138  
372             1810        6929  
...              ...         ...  
20165           2910        5260  
20597           1330        1466  
20654           2680        5992  
20764           1130        1598  
21565           2850        6000  

[177 rows x 21 columns]